paper/paper.md

```diff
@@ -0,0 +1,218 @@
+
+### Slicing
+
+Similar to `std::mdspan`, DDC containers support slicing through the bracket operator that always returns a `ChunkSpan`. The key property is that the resulting slice’s set of `DiscreteElement` is a subset of the original set.
+
+### Multidimensional algorithms
+
+Finally, DDC offers multidimensional algorithms to manipulate the containers and associated `DiscreteElement` indices. The parallel versions are based on Kokkos providing performance portability. Here is a list as of version 0.5:
+
+- `parallel_deepcopy`, copies two `ChunkSpan` that are defined on the same set of `DiscreteElement`,
+- `parallel_for_each`, applies a function once per `DiscreteElement` of a set of `DiscreteElement`,
+- `parallel_fill`, fills a `ChunkSpan` with a value,
+- `parallel_transform_reduce`, applies a function once per `DiscreteElement` of a set of `DiscreteElement` and combines the returned values with a reducer.
+
+## Example
+
+Let us illustrate the basic concepts introduced above with an example that initializes an array with a given value.
+
+```cpp
+#include <iostream>
+
+#include <ddc/ddc.hpp>
+
+#include <Kokkos_Core.hpp>
+
+// Define two independent dimensions
+struct Dim1
+{
+};
+struct Dim2
+{
+};
+
+// For the purpose of the demonstration, this function makes only sense with Dim2
+int sum_over_dim2(ddc::ChunkSpan<int, ddc::DiscreteDomain<Dim2>> slice)
+{
+    int sum = 0;
+    for (ddc::DiscreteElement<Dim2> idx2 : slice.domain()) {
+        sum += slice(idx2);
+    }
+    return sum;
+}
+
+int main()
+{
+    // Initialize Kokkos runtime (automatically finalized at the end of scope)
+    Kokkos::ScopeGuard const kokkos_scope;
+
+    // Initialize DDC runtime (automatically finalized at the end of scope)
+    ddc::ScopeGuard const ddc_scope;
+
+    // Create a discrete domain for Dim1 with 5 elements
+    ddc::DiscreteDomain<Dim1> const dom1
+            = ddc::init_trivial_space<Dim1>(ddc::DiscreteVector<Dim1>(5));
+
+    // Create a discrete domain for Dim2 with 7 elements
+    ddc::DiscreteDomain<Dim2> const dom2
+            = ddc::init_trivial_space<Dim2>(ddc::DiscreteVector<Dim2>(7));
+
+    // Define a 2D discrete domain combining Dim1 and Dim2
+    ddc::DiscreteDomain<Dim1, Dim2> const dom(dom1, dom2);
+
+    // Create a 2D array (Chunk) to store integer values on the combined domain
+    ddc::Chunk my_array("my_array", dom, ddc::HostAllocator<int>());
+
+    // Iterate over the first dimension (Dim1)
+    for (ddc::DiscreteElement<Dim1> const idx1 : dom1) {
+        // Iterate over the second dimension (Dim2)
+        for (ddc::DiscreteElement<Dim2> const idx2 : dom2) {
+            // Assign the value 1 to each element in the 2D array
+            my_array(idx1, idx2) = 1;
+
+            // The following would NOT compile as my_array expects a DiscreteElement over Dim2
+            // my_array(idx1, idx1) = 1;
+        }
```

Comment:
Long line doesn't display correctly
![image](https://github.com/user-attachments/assets/b2b407f5-6666-4bfd-ada8-e0611fef3ae8)

Reply:
I will fix this

Reply:
@tpadioleau doesn't seem to have been fixed

Reply:
working on this tomorrow

Reply:
Confirmed that it is now on the page (although into the margin)

![image](https://github.com/user-attachments/assets/83fe2375-a002-41c9-a321-5034204e3c2a)